go/libraries/utils/argparser/parser.go

```diff
@@ -281,90 +308,103 @@ func (ap *ArgParser) Parse(args []string) (*ArgParseResults, error) {
 	i := 0
 	for ; i < len(args); i++ {
 		arg := args[i]
-		isLongFormFlag := len(arg) >= 2 && arg[:2] == "--"
 
 		if len(arg) == 0 || arg[0] != '-' || arg == "--" { // empty strings should get passed through like other naked words
 			list = append(list, arg)
 			continue
 		}
 
-		arg = strings.TrimLeft(arg, "-")
+		var err error
+		i, list, results, err = ap.parseToken(args, i, list, results)
 
-		if arg == helpFlag || arg == helpFlagAbbrev {
-			return nil, ErrHelp
+		if err != nil {
+			return nil, err
 		}
+	}
 
-		modalOpts, rest := ap.matchModalOptions(arg)
+	if i < len(args) {
+		copy(list, args[i:])
+	}
 
-		for _, opt := range modalOpts {
-			if _, exists := results[opt.Name]; exists {
-				return nil, errors.New("error: multiple values provided for `" + opt.Name + "'")
-			}
+	if ap.MaxArgs != -1 && len(list) > ap.MaxArgs {
+		return nil, ap.TooManyArgsErrorFunc(list)
+	}
 
-			results[opt.Name] = ""
-		}
+	return &ArgParseResults{results, list, ap}, nil
+}
 
-		opt, value := ap.matchValueOption(rest, isLongFormFlag)
+func (ap *ArgParser) parseToken(args []string, i int, list []string, results map[string]string) (int, []string, map[string]string, error) {
```

Comment:
Can you add names to the output variables.  It's not clear what is returned, and I suspect all you need it to add descriptive names - i -> index, list -> ???, results -> foundArguments.

This is one of these cases where gDoc would be reasonable, but probably not required if you had more descriptive names.

Reply:
Done